app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoRepo.kt

```diff
@@ -39,102 +31,46 @@ open class PocketVideoRepo(
         object Inactive : FeedState()
     }
 
-    private val _feedState = MutableLiveData<FeedState>().apply {
-        // setValue must be called from the main thread, and this repo is instantiated
-        // off of the main thread in Espresso tests, so postValue is used instead
-        postValue(buildConfigDerivables.initialPocketRepoState)
-    }
-    open val feedState: LiveData<FeedState> = _feedState
+    private val _feedState = BehaviorSubject.createDefault(initialState)
+    open val feedState = _feedState.hide()
+        .observeOn(AndroidSchedulers.mainThread())
 
-    @Suppress("ANNOTATION_TARGETS_NON_EXISTENT_ACCESSOR") // Private properties generate fields so method annotations can't apply.
-    @get:UiThread
-    @set:UiThread
-    private var backgroundUpdates: Job? = null
-    @Volatile private var lastSuccessfulUpdateMillis = -1L
-    @Volatile private var lastUpdateAttemptMillis = -1L
-    @Volatile private var retryDelayMillis = BASE_RETRY_TIME
-    @Volatile private var lastAttemptWasSuccessful = false
+    private val periodicRequester = PeriodicRequester(pocketEndpoint)
+    private val compositeDisposable = CompositeDisposable()
 
     fun update() {
-        retryDelayMillis = BASE_RETRY_TIME
-        GlobalScope.launch { updateInner() }
+        pocketEndpoint.request()
+            .subscribeOn(Schedulers.io())
+            .subscribe(this::postUpdate)
+            .addTo(compositeDisposable)
     }
 
     @UiThread // update backgroundUpdates.
     fun startBackgroundUpdates() {
-        backgroundUpdates?.cancel() // Cancelling unexpectedly active Pocket update job to ensure only one is running
-        backgroundUpdates = startBackgroundUpdatesInner()
+        compositeDisposable.clear()
+        periodicRequester.start()
+            .subscribe(this::postUpdate)
+            .addTo(compositeDisposable)
     }
 
     // When we the app is not in use, we don't want to hit the network for no reason, so we cancel updates
     @UiThread // stop updating backgroundUpdates.
     fun stopBackgroundUpdates() {
-        backgroundUpdates?.cancel()
-        backgroundUpdates = null
+        compositeDisposable.clear()
```

Comment:
Is this asynchronous?

Reply:
Nope, it's synchronous.  Most Rx is synchronous by default, although there are a few exceptions.